nearbridge/contracts/NearBridge.sol

```diff
@@ -247,70 +246,73 @@ contract NearBridge is INearBridge {
-            }
-            nextBlockProducers.totalStake = totalStake;
+    function setBlockInfo(
+        NearDecoder.LightClientBlock memory src,
+        BlockInfo storage destBlock,
+        BlockProducerInfo storage destBPs
+    )
+        internal
+    {
+        destBlock.height = src.inner_lite.height;
+        destBlock.timestamp = src.inner_lite.timestamp;
+        destBlock.epochId = src.inner_lite.epoch_id;
+        destBlock.nextEpochId = src.inner_lite.next_epoch_id;
+        destBlock.hash = src.hash;
+        destBlock.merkleRoot = src.inner_lite.block_merkle_root;
+        destBlock.next_hash = src.next_hash;
+        destBlock.approvals_after_next_length = src.approvals_after_next.length;
+        for (uint i = 0; i < src.approvals_after_next.length; i++) {
+            destBlock.approvals_after_next[i] = src.approvals_after_next[i];
         }
 
-        if (!init) {
-            // Backup the head. No need to backup if it is initialization.
-            backupHead = head;
-            for (uint i = 0; i < head.approvals_after_next_length; i++) {
-                backupHead.approvals_after_next[i] = head.approvals_after_next[i];
+        if (src.next_bps.none) {
+            destBPs.bpsLength = 0;
+            destBPs.totalStake = 0;
+        } else {
+            destBPs.bpsLength = src.next_bps.validatorStakes.length;
+            uint256 totalStake = 0;
+            for (uint i = 0; i < src.next_bps.validatorStakes.length; i++) {
+                destBPs.bps[i] = BlockProducer({
+                    publicKey: src.next_bps.validatorStakes[i].public_key,
+                    stake: src.next_bps.validatorStakes[i].stake
+                });
+                totalStake = totalStake.add(src.next_bps.validatorStakes[i].stake);
             }
+            destBPs.totalStake = totalStake;
         }
 
-        // Update the head.
-        head = BlockInfo({
-            height: nearBlock.inner_lite.height,
-            epochId: nearBlock.inner_lite.epoch_id,
-            nextEpochId: nearBlock.inner_lite.next_epoch_id,
-            submitter: msg.sender,
-            validAfter: init ? 0 : block.timestamp.add(lockDuration),
-            hash: keccak256(data),
-            next_hash: nearBlock.next_hash,
-            approvals_after_next_length: nearBlock.approvals_after_next.length
-        });
-        for (uint i = 0; i < nearBlock.approvals_after_next.length; i++) {
-            head.approvals_after_next[i] = nearBlock.approvals_after_next[i];
+        emit BlockHashAdded(
+            src.inner_lite.height,
+            src.hash
+        );
+    }
+
+    function commitBlock() internal {
+        require(lastValidAt != 0 && block.timestamp >= lastValidAt, "Nothing to commit");
+
+        head = untrustedHead;
+        if (untrustedHeadIsFromNextEpoch) {
+            copyBlockProducers(nextBlockProducers, currentBlockProducers);
+            copyBlockProducers(untrustedNextBlockProducers, nextBlockProducers);
```

Comment:
Please add a comment saying that if block is from the next epoch it is required to have next block producer so `nextBlockProducers` will never be overridden with empty collection.